docs/operation/upgrade.rst

```diff
@@ -0,0 +1,94 @@
+Upgrade Guide
+=============
+This document describes a complete and reliable upgrade path for **MetalK8s**
+and all the components that make up the stack.
+
+.. important::
+
+    - Upgrading MetalK8s is a manual process and as a result,
+      the following outlined steps must be met to the later in other to achieve
+      a successful upgrade.
+    - The output of the below-provided commands solely depends on the particular
+      MetalK8s setup you have in place.
+
+.. warning::
+    For any cluster upgrade from MetalK8s v2.0 to v2.1 and on-ward, **Kubelet**
+    must be upgraded before making any upgrades to the salt-master. Once this
+    is done, an upgrade is guaranteed to proceed in a seamless manner.
+
+Supported Versions
+******************
+
+This guide currently supports MetalK8s upgrades for the following versions.
+Please refer to the release notes for more information.
+
+    - Upgrade 2.0.0 to 2.1.0
+    - Upgrade 2.1.0 to 2.2.0
```

Comment:
I think you should just say that we support upgrade and downgrade only from 1 minor and give an example but not a full list. Because in the futur (maybe really soon) we will have 2.3.0 2.4.0 ... And you can also ugprade from 2.0.0 to 2.0.1 btw but at most to 1 minor

Reply:
Well, this to me looks like current documentation which will be updated when the next releases of Metalk8s comes out. This is the reason why i listed all the supported upgrades.

We also want users to be sure that their upgrade is supported and this should be a reference point + release notes ... No??

Reply:
Btw in the upgrade orchestrate we check that we have at most 1 minor between current version and destination version.
Hmm if we really want to list them here we should only have
```
Upgrade from 1.x to 2.0.x
Downgrade from 2.1.x to 2.0.x
```
And then in development/2.1 you will have
```
Upgrade from 2.0.x to 2.1.x
Downgrade from 2.2.x to 2.1.x
```

Honnestly we don't want to maintain it (at least me I don't :stuck_out_tongue: )

This section should be global IMO even if we will maybe in another version update it for me we should do it as simple as possible. We just describe the supported path